src/include/rear/ui.rb

```diff
@@ -441,25 +473,55 @@ def RearConfigDialog
           "<p><b>OK</b> saves the configuration and quits while <b>Cancel</b> closes the configuration dialog without saving.<p>"
         )
 
-
       # get varibales from config
       netfs_url = [Rear.netfs_url]
       netfs_keep_old_backup = Rear.netfs_keep_old_backup
-      modules_load = deep_copy(Rear.modules_load)
-      backup_prog_include = deep_copy(Rear.backup_prog_include)
+      use_dhclient = Rear.use_dhclient
+      modules_load = Rear.modules_load
+      backup_prog_include = Rear.backup_prog_include || []
+      post_recovery_script = Rear.post_recovery_script || []
+      required_progs = Rear.required_progs || []
+      copy_as_is = Rear.copy_as_is || []
+      backup_options = Rear.backup_options.empty? ? BACKUP_OPTIONS : Rear.backup_options
       output = Rear.output
 
+      # set defaults:
+      if RearSystemCheck.Btrfs?
+        # Add the important elements to the existing config array:
+        backup_prog_include |= GetMountpoints()
+      end
+
+      # same here
+      required_progs |= REQUIRED_PROGS
+      copy_as_is |= COPY_AS_IS
+      post_recovery_script |= POST_RECOVERY_SCRIPT
```

Comment:
ok, we need to be explicit about it. E.g. consider that user open dialog and see some configuration, so he said: "OK, everything is correct, so I press cancel" and result will be that on system something can missing because we add it only in UI.